pkgs/development/python-modules/taichi/default.nix

```diff
@@ -0,0 +1,250 @@
+    echo "Vulkan: ${builtins.toString enableVulkan}"
+
+    echo "Platform libs: ${builtins.toString platformLibs}"
+
+    git init
+    git config user.email "you@example.com"
+    git commit -m "initial commit" --allow-empty
+    git describe --match=DummyTagNotExisting --always --abbrev=40 --dirty
+    # echo "${src.gitRepoUrl}"
+    # git clone --no-checkout --depth 1 --branch ${src.tag} ${src.gitRepoUrl} ./tmp
+    # cp -t ./tmp/.git .
+    # rm -rf ./tmp
+  '';
+
+  taichi =
+    # let
+    # in
+    # If the flag is a boolean, we'll pass "ON" or "OFF" to CMake.
+    # Otherwise, we'll pass the value of the flag.
+    stdenv.mkDerivation {
+      inherit
+        pname
+        version
+        src
+        cmakeFlags
+        preConfigure
+        ;
+
+      propagatedNativeBuildInputs = [
+        autoPatchelfHook
+        cmake
+        pybind11
+        git
+      ] ++ lib.optionals cudaSupport [ autoAddDriverRunpath ];
+
+      propagatedBuildInputs = [
+        elfutils
+        libbfd
+        libdwarf
+        libdwarf.dev
+        libffi.dev
+        libxml2.dev
+
+        libX11
+        libXrandr
+        libXinerama
+        libXcursor
+        libXi
+
+        libllvm
+      ] ++ platformLibs;
+
+      # See https://github.com/NixOS/nixpkgs/issues/144170
+      # Taken from original PR,
+      # maybe this is more future proof:
+      # https://github.com/NixOS/nixpkgs/issues/144170#issuecomment-1423195220
+      postPatch = ''
+        substituteInPlace \
+          external/SPIRV-Headers/SPIRV-Headers.pc.in \
+          external/SPIRV-Cross/pkg-config/spirv-cross-c-shared.pc.in \
+          external/SPIRV-Tools/cmake/SPIRV-Tools.pc.in \
+          external/SPIRV-Tools/cmake/SPIRV-Tools-shared.pc.in \
+          --replace-warn '$'{prefix}/@CMAKE_INSTALL_LIBDIR@ @CMAKE_INSTALL_FULL_LIBDIR@  \
+          --replace-warn '$'{prefix}/@CMAKE_INSTALL_INCLUDEDIR@ @CMAKE_INSTALL_FULL_INCLUDEDIR@
+      '';
+
+      doCheck = true;
+      nativeCheckInputs = [ gtest ];
+    };
+in
+buildPythonPackage {
+  inherit
+    pname
+    version
+    src
+    cmakeFlags
+    stdenv
+    preConfigure
+    ;
+
+  nativeBuildInputs =
+    [
+      ninja
+      numpy
+      scikit-build
+      setuptools
+      wheel
+    ]
+    ++ lib.optionals (stdenv.hostPlatform == stdenv.buildPlatform) [
+      pythonImportsCheckHook
+      pythonCatchConflictsHook
+    ];
+
+  buildInputs = [
+    taichi
+  ];
+
+  dependencies = [
+    colorama
+    dill
+    matplotlib
+    numpy
+    rich
+  ];
+
+  postConfigure = ''
+    cd ..
+  '';
+
+  # This is needed for the libraries like opengl and vulkan to be found at runtime
+  preInstallCheck = ''
+    shopt -s globstar
+
+    for file in $out/**/*.so; do
+      patchelf --add-rpath "${lib.makeLibraryPath platformLibs}" "$file"
+    done
+  '';
+
+  # dontUseCmakeConfigure = true;
+  # dontUseNinjaBuild = true;
+  # dontUseNinjaInstall = true;
+  dontUseNinjaCheck = true;
+
+  doCheck = true;
```

Comment:
this is default

Reply:
I was checking something and it said it was off in the repl, but I guess it is on during the build time!